### PR TITLE
Migrate JVM-SDK from 1.60 to 1.62

### PR DESCRIPTION
**Pull request by ahmetoz**
We need to upgrade the version to the latest and fix the workarounds in this PR.

For release notes, refer: http://commercetools.github.io/commercetools-jvm-sdk/apidocs/io/sphere/sdk/meta/ReleaseNotes.html#v1_62_0

Review tips:

PR is based on the next major release.
Check commits separately as changes are not related to each other.

### CODE REVIEW

File: src/main/java/com/commercetools/sync/services/impl/BaseServiceWithKey.java [hunk 1]
```diff
@@ -22,6 +23,8 @@
  *     io.sphere.sdk.categories.CategoryDraft}, etc..
  * @param <U> Resource (e.g. {@link io.sphere.sdk.products.Product}, {@link
  *     io.sphere.sdk.categories.Category}, etc..
+ * @param <UQ> Resource to be used on queries (e.g. {@link
```
Comment (lojzatran) [inline src/main/java/com/commercetools/sync/services/impl/BaseServiceWithKey.java:26]: Maybe like this?
```suggestion
 * @param <UQ> Resource returned by the query <Q> (e.g. {@link
```

File: src/main/java/com/commercetools/sync/services/impl/ProductServiceImpl.java [hunk 2]
```diff
@@ -117,104 +87,61 @@ public ProductServiceImpl(@Nonnull final ProductSyncOptions syncOptions) {
...
+    return updateResourceAndMapToProjection(productProjection, updateActions);
   }
-  // TODO: (JVM-SDK), remove this replicated code, when the the JVM-SDK allows to use
-  // ProductProjections in a ProductUpdateCommand.
 
-  // see: https://github.com/commercetools/commercetools-sync-java/pull/699#discussion_r600457980
   @Nonnull
-  CompletionStage<ProductProjection> updateResourceByKey(
-      @Nonnull final ProductProjection resource,
+  CompletionStage<ProductProjection> updateResourceAndMapToProjection(
```
Comment (lojzatran) [inline src/main/java/com/commercetools/sync/services/impl/ProductServiceImpl.java:128]: Where in the code is this part `MapToProjection`?

File: src/main/java/com/commercetools/sync/products/utils/ProductUpdateActionUtils.java [hunk 3]
```diff
@@ -912,21 +910,10 @@ private static boolean isSameForAllActionNew(
     final UpdateAction<Product> addVariant =
         AddVariant.of(draft.getAttributes(), draft.getPrices(), draft.getSku(), true)
             .withKey(draft.getKey())
-            .withImages(draft.getImages());
+            .withImages(draft.getImages())
+            .withAssetDrafts(draft.getAssets());
 
     actions.add(addVariant);
```
Comment (praveenkumarct) [inline src/main/java/com/commercetools/sync/products/utils/ProductUpdateActionUtils.java:916]: Do you still need this actions list?

**Review by lojzatran — CHANGES_REQUESTED** [review]

File: src/main/java/com/commercetools/sync/products/utils/ProductUpdateActionUtils.java (hunk 3 above)
Comment (ahmetoz) [reply src/main/java/com/commercetools/sync/products/utils/ProductUpdateActionUtils.java]: https://github.com/commercetools/commercetools-sync-java/pull/714/commits/a737dbf424865929542715f3c2d76cda4d0502b8

File: src/main/java/com/commercetools/sync/services/impl/ProductServiceImpl.java (hunk 2 above)
Comment (ahmetoz) [reply src/main/java/com/commercetools/sync/services/impl/ProductServiceImpl.java]: https://github.com/commercetools/commercetools-sync-java/pull/714/commits/849dd8aba172d4cd02a138e6dfd75635f3fa80c3
check `.thenApply(p -> p.toProjection(STAGED)));`

File: src/main/java/com/commercetools/sync/services/impl/BaseServiceWithKey.java (hunk 1 above)
Comment (ahmetoz) [reply src/main/java/com/commercetools/sync/services/impl/BaseServiceWithKey.java]: https://github.com/commercetools/commercetools-sync-java/pull/714/commits/2cfa0e15c3390988e53774876b6ea29f6af58669

**Review by lojzatran — APPROVED** [review]

**Pull request merged**